src/foundry/client-esm/nue/tour.d.mts

```diff
@@ -1,9 +1,276 @@
+import type { ValueOf } from "fvtt-types/utils";
+
 /**
  * A Tour that shows a series of guided steps.
- * @remarks TODO: Stub
  */
-declare class Tour {}
+declare class Tour {
+  /**
+   * Construct a Tour by providing a configuration.
+   * @param config  - The configuration of the Tour
+   * @param options - Additional options for configuring the tour
+   *                  (default: `{}`)
+   */
+  constructor(config: Tour.Config, options?: Tour.ConstructorOptions);
+
+  static STATUS: Readonly<{
+    UNSTARTED: "unstarted";
+    IN_PROGRESS: "in-progress";
+    COMPLETED: "completed";
+  }>;
+
+  /**
+   * Indicates if a Tour is currently in progress.
+   */
+  static get tourInProgress(): boolean;
+
+  /**
+   * Returns the active Tour, if any
+   */
+  static get activeTour(): Tour | null;
+
+  /**
+   * Handle a movement action to either progress or regress the Tour
+   * @param movementDirections - The Directions being moved in
+   */
+  static onMovementAction(movementDirections: string[]): boolean | void;
+
+  /**
+   * Configuration of the tour. This object is cloned to avoid mutating the original configuration.
+   */
+  config: Tour.Config;
+
+  /**
+   * The HTMLElement which is the focus of the current tour step.
+   */
+  targetElement: HTMLElement | undefined | null;
+
+  /**
+   * The HTMLElement that fades out the rest of the screen
+   */
+  fadeElement: HTMLElement | undefined;
+
+  /**
+   * The HTMLElement that blocks input while a Tour is active
+   */
+  overlayElement: HTMLElement | undefined;
+
+  /**
+   * Padding around a Highlighted Element
+   * @defaultValue `10`
+   */
+  static HIGHLIGHT_PADDING: number;
+
+  /**
+   * The unique identifier of the tour
+   */
+  get id(): string | undefined;
+
+  set id(value: string | undefined);
```

Comment:
Foundry documents `string` and runtime seems to match. `get id` returns `this.#id` and `this.#id` is, I believe, always set in the constructor. Specifically in the constructor `this.#id = id ?? config.id;` and `config.id` appears to be typed as required.

However that does make this code a bit odd:
```js
  set id(value) {
    if ( this.#id ) throw new Error("The Tour has already been assigned an ID");
    this.#id = value;
  }
```

Perhaps it's meant for the case of `id === ""` alone?